src/mmw/js/src/core/utils.js

```diff
@@ -0,0 +1,43 @@
+"use strict";
+
+var _ = require('underscore'),
+    md5 = require('blueimp-md5').md5;
+
+var utils = {
+
+    getCollectionHash: function(collection, sortKey) {
+        // JSON objects are not guaranteed consistent order in string
+        // representation, so we convert to sorted array which can be
+        // stringified reliably.
+
+        // Converts an object with JSON representation:
+        //     {name: 'abc', type: 'xyz', shape: [...]}
+        // to:
+        //     [{name: 'abc'}, {shape: [...]}, {type: 'xyz'}]
```

Comment:
Yes. A modification, for example, has this structure:

```json
    {
        "area": 648.1614728140878,
        "name": "landcover",
        "shape": {
            "geometry": {
                "coordinates": [
                    [
                        [
                            -75.1647502183914,
                            39.95236473500341
                        ],
                        [
                            -75.16454637050629,
                            39.95232772476147
                        ],
                        [
                            -75.16461610794067,
                            39.952011080761736
                        ],
                        [
                            -75.1648199558258,
                            39.952031642104906
                        ],
                        [
                            -75.1647502183914,
                            39.95236473500341
                        ]
                    ]
                ],
                "type": "Polygon"
            },
            "properties": {},
            "type": "Feature"
        },
        "type": "",
        "units": "m<sup>2</sup>",
        "value": "grassland"
    }
```
That's why we recursively sort nested objects in lines [22-25](https://github.com/WikiWatershed/model-my-watershed/pull/567/files#diff-8b62826bd98f0674cd1f97e9bb645f8cR22) below.